compiler-rt/lib/ctx_profile/CtxInstrProfiling.h

```diff
@@ -113,6 +113,28 @@ struct ContextRoot {
   static_assert(sizeof(Taken) == 1);
 };
 
+// This is allocated and zero-initialized by the compiler, the in-place
+// initialization serves mostly as self-documentation and for testing.
+// The design is influenced by the observation that typically (at least for
+// datacenter binaries, which is the motivating target of this profiler) less
+// than 10% of functions in a binary even appear in a profile (of any kind).
+//
+// 1) We could pre-allocate the flat profile storage in the compiler, just like
+// the flat instrumented profiling does. But that penalizes the static size of
```

Comment:
Zero initialized storage in .bss shouldn't increase binary size significantly though?

Reply:
right but then the upfront memory requirement would go up - see point 2 below. added the ".bss" bit, that wasn't clear.